packages/api/core/src/api/init-scripts/init-link.ts

```diff
@@ -27,31 +26,113 @@ export async function initLink<T>(
   if (shouldLink) {
     d('Linking forge dependencies');
     const packageJson = await readRawPackageJson(dir);
-    // TODO(erickzhao): the `--link-folder` argument only works for `yarn`. Since this command is
-    // only made for Forge contributors, it isn't a big deal if it doesn't work for other package managers,
-    // but we should make it cleaner.
-    const linkFolder = path.resolve(
+    const forgeRoot = path.resolve(
       __dirname,
       '..',
       '..',
       '..',
       '..',
       '..',
       '..',
-      '.links',
     );
+
```

Comment:
to resolve linking issues, we must custom retrieve each workspace's path and link them with yarn 4 syntax